pinot-spi/src/main/java/org/apache/pinot/spi/utils/IngestionConfigUtils.java

```diff
@@ -116,7 +117,7 @@ public static String getBatchSegmentIngestionType(TableConfig tableConfig) {
     if (segmentIngestionType == null) {
       segmentIngestionType = tableConfig.getValidationConfig().getSegmentPushType();
     }
-    return segmentIngestionType;
+    return (segmentIngestionType == null) ? DEFAULT_SEGMENT_INGESTION_TYPE : segmentIngestionType;
```

Comment:
what was the motivation for this change? Why is `APPEND` the right default vs throwing an exception? We've encountered a pretty large issue where misconfigured tables had segment names change

Reply:
I don't remember the history of this change, but this change seems not related to this PR. Which code path is causing the incident?

Reply:
Defaulting to `APPEND` causes https://github.com/apache/pinot/blob/master/pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/name/NormalizedDateSegmentNameGenerator.java#L78 to switch to the `if (_appendPushType)` which renames all segments. Previous, it went down the other branch first. For refresh tables, this led to duplicated data.

I'm not sure there's a good default here. It should just throw an exception. But nothing validates ingestion type or frequency from what I can tell.

Reply:
Throwing exception could potentially break existing use cases.

The push type should be either APPEND or REFRESH, and there are other places where we check whether the push type is REFRESH, e.g. https://github.com/apache/pinot/blob/master/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/ConsistentDataPushUtils.java#L221. We shouldn't allow empty push type which doesn't match APPEND or REFRESH.

Since we are already making it default to `APPEND`, and this has been merged for 1 year, I guess we need to follow this going forward. I'm labelling this PR backward incompatible. Majority of the use cases should be `APPEND` when the time column is configured.

Sorry for causing this incident. Will also update the documentation about `APPEND` being the default push type

Reply:
> We shouldn't allow empty push type which doesn't match APPEND or REFRESH.

Agreed. Though it's not clear to me where we would put this logic since REALTIME table configs are ok with this being unset.

> Since we are already making it default to APPEND, and this has been merged for 1 year, I guess we need to follow this going forward.

Looks like this PR added a unit test in `pinot-spi/src/test/java/org/apache/pinot/spi/utils/IngestionConfigUtilsTest.java` to assert the default is now `APPEND`. So this is probably fine going forward.